src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/TypeChecking/Data/Inference.hs

```diff
@@ -166,7 +214,7 @@ strongNormalize' = go
     goApp (Application l r i) = do
       l' <- go l
       case l' of
-        ExpressionSimpleLambda (SimpleLambda lamVar _ lamBody) -> do
+        ExpressionSimpleLambda (SimpleLambda (SimpleBinder lamVar _) lamBody) -> do
           go (substitutionE (HashMap.singleton lamVar r) lamBody)
```

Comment:
I'm not sure if this is correct. What if `r` contains some binders, e.g., lets or lambdas? They could be duplicated by the substitution, which would violate the assumption that name ids are unique for different binders. I think either there should be an explicit assumption here that `r` contains no binders (which should be guaranteed / checked somehow) or the bound variables in `r` should be refreshed.

Reply:
Your are right, I've had this on the back on my mind for some time, but since it wasn't causing problems I didn't change it. Probably because SimpleLamba is only used for type synonyms, and we haven't used nested type synonyms applied to arguments with binders.
However, this pr does not change the behaviour of this, I merely adopted the syntax to use the new SimpleBinder constructor.

Reply:
Okay, let's do it in a separate PR